privacyidea/lib/eventhandler/responsemangler.py

```diff
@@ -0,0 +1,176 @@
+                             },
+                        "type":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The type of the value."),
+                             "value": ["string", "integer", "bool"]
+                            },
+                        "value":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The value of the JSON key that should be added.")
+                            }
+                        }
+                   }
+        return actions
+
+    def do(self, action, options=None):
+        """
+        This method executes the defined action in the given event.
+
+        :param action:
+        :param options: Contains the flask parameters g, request, response
+            and the handler_def configuration
+        :type options: dict
+        :return:
+        """
+        ret = True
+        response = options.get("response")
+        content = self._get_response_content(response)
+        handler_def = options.get("handler_def")
+        handler_options = handler_def.get("options", {})
+
+        json_pointer = handler_options.get("JSON pointer", "")
+        value = handler_options.get("value")
+        type = handler_options.get("type")
+        comp = json_pointer.split("/")
+        comp = [x for x in comp if x]
+        if action.lower() == ACTION_TYPE.DELETE:
+            try:
+                if len(comp) == 1:
+                    del(content[comp[0]])
+                elif len(comp) == 2:
+                    del(content[comp[0]][comp[1]])
+                elif len(comp) == 3:
+                    del (content[comp[0]][comp[1]][comp[2]])
+                else:
+                    log.warning("JSON pointer length of {0!s} not supported.".format(len(comp)))
+                options.get("response").data = json.dumps(content)
+            except KeyError:
+                log.warning("Can not delete response JSON Pointer {0!s}.".format(json_pointer))
+        elif action.lower() == ACTION_TYPE.ADD:
+            try:
+                if type == "integer":
+                    value = int(value)
+                elif type == "bool":
+                    value = is_true(value)
+            except ValueError:
+                log.warning("Failed to convert value")
+            try:
+                if len(comp) == 1:
+                    content[comp[0]] = value
+                elif len(comp) == 2:
+                    content[comp[0]] = content.get(comp[0]) or {}
+                    content[comp[0]][comp[1]] = value
+                elif len(comp) == 3:
+                    content[comp[0]] = content.get(comp[0]) or {}
+                    content[comp[0]][comp[1]] = content[comp[0]].get(comp[1]) or {}
+                    content[comp[0]][comp[1]][comp[2]] = value
```

Comment:
I checked some things with python modules like ``jsonpatch`` but they all usually have the problem, that they do not "create" the missing path.

~~~~python
jp = jsonpatch.JsonPatch([{'op': 'add', 'path': '/detail/message', 'value': "Hallo"}])
original_dict = {"original": "dict"}
jp.apply(original_dict)
~~~~
In this example I would like to add a pointer ``/detail/message`` to the "original_dict". But if the path to this pointer does not exist, yet, it will bail out with an error.
So even when using external modules, we would have to create the missing path manually which creates some nasty code.